src/models/event.ts

```diff
@@ -285,7 +285,7 @@ export class MatrixEvent extends EventEmitter {
      * @return {string} The event type, e.g. <code>m.room.message</code>
      */
     public getType(): EventType | string {
-        return this.clearEvent.type || this.event.type;
+        return this.clearEvent?.type || this.event.type;
```

Comment:
this falling-through for a decrypted event with malformed `type` in the encrypted payload is not ideal, it'll behave inconsistently where getType() may be for wire or payload and getContent() for the other